### DISCUSSION TITLE
How to deploy remark42 on vue3?

### DISCUSSION THREAD

**Discussion by renjunok** [umputun/remark42#1551] [General]
I've been trying for 3 days and would appreciate any help available, thanks!

**Comment by paskal** [comment]
Hi @renjunok, can you please describe what do you have working and what is the problem you have? Do you have a remark42 instance working with /web/ endpoint, looking similar to https://remark42.com/demo/ so that you can leave a comment, refresh the page and still see it?

**Reply by renjunok** [reply]
Hi, appreciate your reply!
I am a beginner of vue, refer to https://remark42.com/docs/configuration/frontend/spa/
Install remark42, now the statement reports an error, I have taken a screenshot

<img width="1514" alt="image" src="https://user-images.githubusercontent.com/10955851/205655436-040a8407-c657-494e-8b39-43dbdd6ae090.png">

Also is there a code sample for vue3 route for the code below?
```@Watch('$route. path')
   onRouteChange() {
     this.initRemark42()
   }
```

Thank you!

**Reply by akellbl4** [reply]
As I understand you don't have ref to the element. Could you please post code of the whole file or maybe you have in on github?

**Reply by renjunok** [reply]
This is index.html code  a screenshot

<img width="920" alt="image" src="https://user-images.githubusercontent.com/10955851/205827201-9f5fb244-db29-4bba-a8cf-38e5c3657f7a.png">

This is vue file
<img width="1229" alt="image" src="https://user-images.githubusercontent.com/10955851/205830203-9dbe6319-970a-4e09-ada4-63d0192cde05.png">

**Reply by akellbl4** [reply]
You don't have ref assigned to remark div. Check vue docs https://vuejs.org/guide/essentials/template-refs.html#template-refs

**Reply by renjunok** [reply]
Thank you, I have made a reference to remark42 according to the vue document, but still report an error.
<img width="533" alt="image" src="https://user-images.githubusercontent.com/10955851/205974540-02944494-827a-4504-b2c8-cee821dd1fbc.png">
<img width="1158" alt="image" src="https://user-images.githubusercontent.com/10955851/205974727-0f322515-6191-4bd1-a8c7-77065d78a893.png">

error line:  this.$refs.remark42 as HTMLElement,
<img width="1103" alt="image" src="https://user-images.githubusercontent.com/10955851/205975014-dd964573-3a03-4fe1-9ce1-09e5502c5c4e.png">
